### ISSUE TITLE
Menu Bar: Close submenu when press the tab key

### ISSUE THREAD

**Issue by xmicore** [vaadin/flow-components#3221]
### Description

When the menubar has an open submenu and the user navigates to the next element via tab the menu stays open and the next element is not focused.

### Expected outcome

When the user presses the tab key on an open submenu, the submenu should be closed and the next element in the tab order should be focused.

### Minimal reproducible example

```java
public class HelloWorldView extends HorizontalLayout {

    public HelloWorldView() {

        final MenuBar menuBar = new MenuBar();
        add(menuBar);

        final MenuItem item1 = menuBar.addItem("Item 1");
        item1.getSubMenu().addItem("Item 1.1");
        item1.getSubMenu().addItem("Item 1.2");
        item1.getSubMenu().addItem("Item 1.3");

        final MenuItem item2 = menuBar.addItem("Item 2");
        item2.getSubMenu().addItem("Item 2.1");
        item2.getSubMenu().addItem("Item 2.2");
        item2.getSubMenu().addItem("Item 2.3");

        final Button button = new Button("Mein Button");
        add(button);
    }

}
```

### Steps to reproduce

1. Navigate to a submenu item via keyboard
2. Press the tab key

### Environment

Vaadin version(s): 23.0.9
OS: Windows 10
Browsers: Chrome 101.0.4951.67, Firefox 91.9.0esr

### Browsers

Chrome, Firefox, Edge

**Comment by knoobie** [comment]
@web-padawan sounds like a web-component a11y issue
